### PR TITLE
[DTensor] Registers sharding rule for rms_norm

### PR DESCRIPTION
**Pull request by AaronWang04**
Reduces collective calls in the forward pass from 2 to 1

In #158716 I added the sharding rule for the backward pass but didn't add the forward pass as it didn't get dispatched. After #159324 this should get properly dispatched hence I am adding it now.

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta

### CODE REVIEW

**Comment by iremyux** [comment]
Adding `ciflow/win-arm64` label to trigger Windows Arm64 CI.
(This is for testing purposes only and isn’t related to any changes in this PR.)

**Comment by AaronWang04** [comment]
cc @XilunWu

File: test/distributed/tensor/test_math_ops.py [hunk 1]
```diff
@@ -271,14 +271,22 @@ def test_layer_norm_fwd(self):
         norm_shape_idx_list = list(range(x.ndim))
         shard_dims = [-1, 0, 1, 2]
         elementwise_affine_list = [False, True]
+
+        # Test RMSNorm as well if CUDA
+        norm_types = [torch.nn.LayerNorm]
+        if self.device_type == "cuda" and hasattr(torch.nn, "RMSNorm"):
+            norm_types.append(torch.nn.RMSNorm)
```
Comment (XilunWu) [inline test/distributed/tensor/test_math_ops.py:278]: this is fine to me but it would be great if we have this kind of thing: https://github.com/pytorch/pytorch/blob/23cf24103963adce84b2b4c027053fec0b29ad94/torch/backends/cuda/__init__.py#L460

or

https://github.com/pytorch/pytorch/blob/23cf24103963adce84b2b4c027053fec0b29ad94/torch/testing/_internal/common_cuda.py#L84

File: test/distributed/tensor/test_math_ops.py [hunk 2]
```diff
@@ -384,7 +403,7 @@ def _replicate_fn(name, module, device_mesh):
             self.assertEqual(
                 sum(comm_mode.comm_module_counts["Global"]["forward"].values()),
                 expected_fwd_comm,
-                f"comm count={comm_mode.get_total_counts()}, "
+                f"comm count={comm_mode.get_total_counts()}, norm_type={norm_type.__name__}, "
```
Comment (XilunWu) [inline test/distributed/tensor/test_math_ops.py:406]: FWIW somehow my PR https://github.com/pytorch/pytorch/pull/159532 also reduces the comm number to 1, which I totally don't know how...

**Review by XilunWu — COMMENTED** [review]
LGTM. Is there a way for us to test this PR's correctness in torchtitan as well? cc @tianyu-l

File: test/distributed/tensor/test_math_ops.py (hunk 2 above)
Comment (AaronWang04) [reply test/distributed/tensor/test_math_ops.py:406]: @XilunWu made a comment on that PR, I think its a separate bug that this PR will hide?

File: test/distributed/tensor/test_math_ops.py (hunk 1 above)
Comment (AaronWang04) [reply test/distributed/tensor/test_math_ops.py:278]: It would be the same `if self.device_type == "cuda"` line of code in that function at least for now so I think it would just be extra abstraction

**Review by tianyu-l — APPROVED** [review]
LGTM, thanks a lot!
